### ISSUE TITLE
Messed up output in gdalwarp

### ISSUE THREAD

**Issue by drnextgis** [OSGeo/gdal#9722]
### What is the bug?

The output lines for `gdalwarp` can become disorganized under certain conditions:

```
$ gdalwarp -tr 100 100 -r cubic test.tif test-cubic-385-nodata.tif
Creating output file that is 3547P x 3581L.
Processing test.tif [1/1] : 0Using internal nodata values (e.g. -2) for image test.tif.
Copying nodata values from source test.tif to destination test-cubic-385-nodata.tif.
...10...20..
```

Expected:
```
Processing test.tif [1/1] : Using internal nodata values (e.g. -2) for image test.tif.
Copying nodata values from source test.tif to destination test-cubic-385-nodata.tif.
0...10...20.
```

### Steps to reproduce the issue

Simply run `gdalwarp`.

### Versions and provenance

```
$ gdalinfo --version
GDAL 3.8.5, released 2024/04/02
```

**Comment by jratike80** [comment]
I guess that gdalwarp really starts to do something else before it is doing "Using internal nodata values (e.g. -2) for image test.tif". Imagine output like this:
 ```
0...10...Doing now something that user may need to know
 20...30...
```

**Comment by drnextgis** [comment]
I see what you mean @jratike80. It's not immediately clear and it makes reading through the progress bar a bit difficult.

**Comment by jratike80** [comment]
A workaround could be to have a dummy space character in the message text.

`0 Using internal nodata values (e.g. -2) for image test.tif.`

**Comment by rouault** [comment]
was already fixed in master per d813ba08564cd34e1175189bf02433c0c53b422f
